src/helpers/utils.ts

```diff
@@ -273,17 +269,16 @@ export function download(file: Blob, fileType: string, fileName: string) {
 }
 
 export function handleStatusRedirect(
-  status: string,
+  status: ApplicationStatus,
+  applicationType: ApplicationType,
   history: RouteComponentProps['history']
 ) {
-  switch (status) {
-    case SUBMITTED:
-      return history.push('/registration-closed?param=validate')
-    case CONFIRMED:
-      return window.location.replace('/home')
-    case DECLINED:
-      return history.push('/registration-closed')
-    default:
-      return history.push('/landing')
-  }
+  if ([ApplicationStatus.SUBMITTED, ApplicationStatus.CONFIRMED, ApplicationStatus.DECLINED].includes(status))
+    location.href = '/'
+  else if (Object.values(ApplicationStatus).includes(status) && !location.search.includes('overlay=consent_osp')) {
+    if (applicationType === ApplicationType.INTERNAL)
+      history.push('/landing')
+    else
+      location.href = '/'
```

Comment:
here I was wrong with my comment earlier - we must specify the parameter because the frontend app checks for it:
`location href = '/?overlay=consent_osp'`